rpa-mapper/src/test/java/uk/gov/hmcts/cmc/rpa/mapper/RequestForJudgementJsonMapperTest.java

```diff
@@ -0,0 +1,135 @@
+@SuppressWarnings({"LineLength"})
+public class RequestForJudgementJsonMapperTest {
+
+    private static final LocalDateTime CCJ_REQUESTED_AT = LocalDate.of(2018, 4, 26).atStartOfDay();
+    private static final LocalDate PAY_BY_SET_DATE = LocalDate.of(2200, 3, 12);
+    private static final BigDecimal PAID_ALREADY = new BigDecimal(10);
+    private static final LocalDate FIRST_PAYMENT_DATE = LocalDate.of(2200, 3, 02);
+    @Autowired
+    private RequestForJudgementJsonMapper mapper;
+
+    @Test
+    public void shouldMapRequestForJudgementForthWith() throws JSONException {
+        SampleCountyCourtJudgment countyCourtJudgment = new SampleCountyCourtJudgment();
+        countyCourtJudgment.withPaidAmount(PAID_ALREADY);
+        Claim claim = SampleClaim.builder()
+            .withCountyCourtJudgmentRequestedAt(CCJ_REQUESTED_AT)
+            .withCountyCourtJudgment(countyCourtJudgment.build())
+            .build();
+        String expected = new ResourceReader().read("/judgement/rpa_request_for_judgement_immediately.json")
+            .trim();
+        assertEquals(expected, mapper.map(claim).toString(), STRICT);
+    }
+
+    @Test
+    public void shouldMapRequestForJudgementImmediatelyButNothingPaid() throws JSONException {
+        SampleCountyCourtJudgment countyCourtJudgment = new SampleCountyCourtJudgment();
+        countyCourtJudgment.withPaidAmount(null);
+        Claim claim = SampleClaim.builder()
+            .withCountyCourtJudgmentRequestedAt(CCJ_REQUESTED_AT)
+            .withCountyCourtJudgment(countyCourtJudgment.build())
+            .build();
+        String expected = new ResourceReader().read("/judgement/rpa_request_for_judgement_immediately_nothing_paid.json")
+            .trim();
+        assertEquals(expected, mapper.map(claim).toString(), STRICT);
+    }
+
+    @Test
+    public void shouldMapRequestForJudgementPaidInFull() throws JSONException {
+        SampleCountyCourtJudgment countyCourtJudgment = new SampleCountyCourtJudgment();
+        countyCourtJudgment.withPaidAmount(PAID_ALREADY);
+        countyCourtJudgment.withPayBySetDate(PAY_BY_SET_DATE);
+        Claim claim = SampleClaim.builder()
+            .withCountyCourtJudgmentRequestedAt(CCJ_REQUESTED_AT)
+            .withCountyCourtJudgment(countyCourtJudgment.build()).build();
+        String expected = new ResourceReader().read("/judgement/rpa_request_for_judgement_full_by_set_date.json")
+            .trim();
+        assertEquals(expected, mapper.map(claim).toString(), STRICT);
+    }
+
+    @Test
+    public void shouldMapRequestForJudgementPayingFortnightly() throws JSONException {
+        RepaymentPlan repaymentPlan = SampleRepaymentPlan
+            .builder()
+            .withPaymentSchedule(PaymentSchedule.EVERY_TWO_WEEKS)
+            .withFirstPaymentDate(FIRST_PAYMENT_DATE)
+            .withInstalmentAmount(BigDecimal.valueOf(100.00)).build();
+        SampleCountyCourtJudgment countyCourtJudgment = new SampleCountyCourtJudgment();
```

Comment:
Please use fluent API since builder offers it to have

```
new SampleCountyCourtJudgment()
  .withPaidAmount(PAID_ALREADY)
  .withRepaymentPlan(repaymentPlan)
```

The same in other places please in the file.

Reply:
Corrected.